app/src/redux-resources/analytics/hooks/useModuleAnalytics.ts

```diff
@@ -0,0 +1,154 @@
+import { useMostRecentCompletedAnalysis } from '/app/resources/runs'
+import {
+    FLEX_ROBOT_TYPE,
+    getDeckDefFromRobotType,
+  } from '@opentrons/shared-data'
+const deckDef = getDeckDefFromRobotType(FLEX_ROBOT_TYPE)
+
+// List of commands that will be recorded
+const commandType = [
+    'thermocycler/closeLid',
+    'thermocycler/openLid',
+    'temperatureModule/setTargetTemperature',
+    'thermocycler/setTargetLidTemperature',
+    'thermocycler/setTargetBlockTemperature',
+    'heaterShaker/setTargetTemperature',
+    'thermocycler/deactivateLid',
+    'thermocycler/deactivateBlock',
+    'temperatureModule/deactivate',
+    'magneticModule/disengage',
+    'heaterShaker/deactivateShaker',
+    'heaterShaker/deactivateHeater',
+    'heaterShaker/openLabwareLatch',
+    'heatershaker/closeLabwareLatch' 
+];
+
+// variables needed to detect find matching modules
+const MODULE_REFETCH_INTERVAL_MS = 5000
+const DECK_CONFIG_POLL_MS = 5000
+
+export interface ReportModuleActionParams {
+    runId?: string | null;
+    moduleType?: string | Dict | null;
+    firmwareVersion?: string | null;
+    action?: string;
+    result?: { status: string| undefined; data: any  }; 
+    errorDetails?: string ; 
+    serialNumber?: string | Dict | null;
+    temperature?: number | null | Dict | string ;
+    params?: Dict | null| undefined;
+}
+
+export interface UseModuleCommandAnalyticsResult {
+    /* Report when a module command completes. */
+    reportModuleCommand: (params: ReportModuleActionParams) => void;
+}
+
+// Define type guards to check the structure of params
+function isModuleOnlyParams(params: any): params is ModuleOnlyParams {
+    return params && typeof params === "object" && "moduleId" in params;
+}
+
+function isTemperatureParams(params: any): params is TemperatureParams {
+    return params && typeof params === "object" && "celsius" in params;
+}
+
+function isParamType(params: any): { moduleId: string; celsius: string } {
+    // Checks param type and returns variables found
+    if (isModuleOnlyParams(params)) {
+        return { moduleId: String(params.moduleId), celsius: "" };
+    }
+    if (isTemperatureParams(params)) {
+        return { moduleId: String(params.moduleId), celsius: String(params.celsius) };
+    }
+    return { moduleId: "", celsius: "" }; // Default return for unexpected cases
+}
+
+export function useModuleCommandAnalytics(modules?: AttachedModule[]): UseModuleCommandAnalyticsResult {
+    const doTrackEvent = useTrackEvent();
+    const reportModuleCommand = ({
+        runId,
+        moduleType,
+        action,
+        result,
+        errorDetails,
+        serialNumber,
+        temperature,
+        params,
+        firmwareVersion
+    }: ReportModuleActionParams & { errorDetails?: string }): void => {
+        if (typeof action !== 'string' || !commandType.includes(action)) {
+            return; // Exit early if action is invalid or not in the commandType array
```

Comment:
```suggestion
            return; 
```

I think a lot of this conditional logic will change as the interface changes, but we don't need a comment for this, since the behavior is obvious to other devs.